runtime/swift-functions.swift

```diff
@@ -0,0 +1,29 @@
+#if os(macOS)
+import AppKit
+#endif
+import Foundation
+import StoreKit
+#if !os(macOS)
+import UIKit
+#endif
+
```

Comment:
I don't know that this matters a lot, but you can simplify this a little bit with a type alias for the platform UI type:

```swift
#if os(macOS)
typealias PlatformScene = NSViewController
#elseif !os(tvOS)
typealias PlatformScene = UIWindowScene
#endif
@objc(XamarinSwiftFunctions)
public class XamarinSwiftFunctions : NSObject {

    @MainActor
    @objc(requestReview:)
    @available(macOS 13, iOS 16, macCatalyst 16, *)
    public static func StoreKit_RequestReview(scene: PlatformScene)
    {
        AppStore.requestReview(in: scene)
    }
}

Reply:
You can also use `#if canImport(UIKit)` instead of a specific platform check around `import` statements.

Reply:
I created a new PR for these changes: #21452.